modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -5280,27 +5292,32 @@ private void _validateExternalReferenceCode(
 	}
 
 	private void _validateFileExtension(
-			String fileExtension, long objectFieldId, String objectFieldName)
+			long dlFileEntryId, String fileExtension, long objectFieldId,
+			String objectFieldName)
 		throws PortalException {
 
 		if (!ArrayUtil.contains(
 				_attachmentManager.getAcceptedFileExtensions(objectFieldId),
 				fileExtension, true)) {
 
+			TempFileEntryUtil.deleteTempFileEntry(dlFileEntryId);
+
```

Comment:
@vngoliveira This seems like new behavior to me.. shouldn't we keep the temporary file and the user should change it in case of failure?